app-drawer/app-drawer.html

```diff
@@ -645,10 +637,16 @@
       },
 
       _openedPersistentChanged: function() {
-        if (this.transitionDuration === 0) {
-          // Reset drawer state now since there will be no transitionend event.
-          this._resetDrawerState();
+        // Don't reset drawer state if tracking or flinging, since _trackEnd and _flingDrawer will
+        // set its own timer.
+        if (this._drawerState === this._DRAWER_STATE.FLINGING ||
+            this._drawerState === this._DRAWER_STATE.TRACKING) {
+          return;
         }
+
+        // Use setTimeout instead of transitionend since transitionend won't fire when app-drawer
+        // is display: none.
+        window.setTimeout(this._resetDrawerState.bind(this), this.transitionDuration);
```

Comment:
I updated this branch to use `debounce` instead of `setTimeout`. Don't think I can use `Polymer.dom.flush()` in the tests though, since those are the places where I'm waiting for the debouncers to be called.